inlang/source-code/versioned-interfaces/language-tag/src/lookup.ts

```diff
@@ -0,0 +1,30 @@
+import type { LanguageTag } from "./interface.js"
+
+/**
+ * Performs a lookup for the given language tag, among the available language tags, 
+ * according to the IETF BCP 47 spec.
+ * 
+ * It **does not support Wildcards** at the moment.
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc4647#section-3.4
+ */
+export function lookup<LanguageTags extends readonly LanguageTag[]>(
+	languageTag: LanguageTags[number],
+	availableLanguageTags: LanguageTags,
+	defaultLanguageTag: LanguageTags[number]
+): LanguageTags[number] {
+	const fallbackLanguages: LanguageTags[number][] = []
+
+	const languageTagParts = languageTag.split("-").filter(Boolean)
+	for (let i = languageTagParts.length; i > 0; i--) {
+		//Skip the x separator
+		if (languageTagParts[i - 1] === "x") continue
+
+		//Stringify the language tag parts
+		const fallbackLanguageTag = languageTagParts.slice(0, i).join("-")
+		if (!availableLanguageTags.includes(fallbackLanguageTag)) continue
+		fallbackLanguages.push(fallbackLanguageTag)
+	}
+
+	return fallbackLanguages[0] ?? defaultLanguageTag
+}
```

Comment:
Should the function throw if the `defaultLanguageTag` is not included in `availableLanguageTags` according to the spec?

Reply:
I don't see the spec mentioning that the default must be part of the options. I don't really see a benefit to enforcing that either. It would make consuming the API a little more awkward.